packages/cip2/test/RoundRobinRandomImprove.test.ts

```diff
@@ -1,237 +1,172 @@
-            utils.createUnspentTxOutput({ coins: 10_000_000n, assets: {} })
-          ],
-          createOutputs: (utils) => [
-            utils.createOutput({ coins: 12_000_000n, assets: {} }),
-            utils.createOutput({ coins: 2_000_000n, assets: {} })
-          ],
-          constraints: NO_CONSTRAINTS,
-          expectedError: InputSelectionFailure.UtxoBalanceInsufficient
+          createUtxo: (utils) => [utils.createUnspentTxOutput({ coins: 3_000_000n })],
+          createOutputs: (utils) => [utils.createOutput({ coins: 3_000_000n })],
+          mockConstraints: NO_CONSTRAINTS
         });
       });
-      it('Coin (Outputs+Fee>UTxO)', async () => {
-        await testInputSelectionFailureMode({
-          getAlgorithm: getRoundRobinRandomImprove,
-          createUtxo: (utils) => [
-            utils.createUnspentTxOutput({ coins: 4_910_000n, assets: {} }),
-            utils.createUnspentTxOutput({ coins: 5_000_000n, assets: {} })
-          ],
-          createOutputs: (utils) => [utils.createOutput({ coins: 10_000_000n, assets: {} })],
-          constraints: {
-            ...NO_CONSTRAINTS,
-            computeMinimumCost: async () => 100_000n
-          },
-          expectedError: InputSelectionFailure.UtxoBalanceInsufficient
+    });
+    describe('Failure Modes', () => {
+      describe('UtxoBalanceInsufficient', () => {
+        it('Coin (Outputs>UTxO)', async () => {
+          await testInputSelectionFailureMode({
+            getAlgorithm: getRoundRobinRandomImprove,
+            createUtxo: (utils) => [
+              utils.createUnspentTxOutput({ coins: 3_000_000n }),
+              utils.createUnspentTxOutput({ coins: 10_000_000n })
+            ],
+            createOutputs: (utils) => [
+              utils.createOutput({ coins: 12_000_000n }),
+              utils.createOutput({ coins: 2_000_000n })
+            ],
+            mockConstraints: NO_CONSTRAINTS,
+            expectedError: InputSelectionFailure.UtxoBalanceInsufficient
+          });
         });
-      });
-      it('Asset', async () => {
-        await testInputSelectionFailureMode({
-          getAlgorithm: getRoundRobinRandomImprove,
-          createUtxo: (utils) => [utils.createUnspentTxOutput({ coins: 10_000_000n, assets: { [TSLA_Asset]: 7000n } })],
-          createOutputs: (utils) => [utils.createOutput({ coins: 5_000_000n, assets: { [TSLA_Asset]: 7001n } })],
-          constraints: NO_CONSTRAINTS,
-          expectedError: InputSelectionFailure.UtxoBalanceInsufficient
+        it('Coin (Outputs+Fee>UTxO)', async () => {
+          await testInputSelectionFailureMode({
+            getAlgorithm: getRoundRobinRandomImprove,
+            createUtxo: (utils) => [
+              utils.createUnspentTxOutput({ coins: 4_910_000n }),
+              utils.createUnspentTxOutput({ coins: 5_000_000n })
+            ],
+            createOutputs: (utils) => [utils.createOutput({ coins: 10_000_000n })],
```

Comment:
```suggestion
            createOutputs: (utils) => [utils.createOutput({ coins: 9_000_000n })],
```
This ensures the presence of a fee exceeds the utxo balance.

Reply:
This test has UTxO cover outputs, but not cover outputs+fee: `10_000_000 - (4_910_000 + 5_000_000) = 90_000` < 100_000 minimumCost, so we should keep this test as it is. Do you want to add an additional test?